src/main/java/core/basesyntax/strategy/impl/PurchaseStorageService.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.StorageService;
+
+public class PurchaseStorageService implements StorageService {
+    @Override
+    public void updateStorage(FruitTransaction transaction) {
+        Storage.fruits
+                .entrySet()
+                .stream()
+                .filter(m -> m.getKey().equals(transaction.getFruit()))
+                .findFirst()
+                .map(m -> m.setValue(m.getValue() - transaction.getQuantity()));
```

Comment:
maybe we can use Map.compute(k, (k, v) -> v + quantity) to not iterate through entryset every time (didn't tried myself)

Reply:
good one